libdnf/common/exception.cpp

```diff
@@ -59,8 +59,13 @@ std::string SystemError::get_error_message() const {
 }
 
 
-std::string format(const std::exception & e, std::size_t level) {
-    std::string ret(std::string(level, ' ') + e.what() + '\n');
+static std::string format(const std::exception & e, std::size_t level) {
+    std::string ret;
+    if (auto ex = dynamic_cast<const Error *>(&e)) {
```

Comment:
I'm not a fan of this notation. I'm not sure if it's now a general consensus, probably not and might be considered personal preference, to be explicit about boolean expressions, in this case comparing against `nullptr` (and then putting the assignment on a separate line, because otherwise it really looks confusing):
```c++
auto ex = dynamic_cast<const Error *>(&e);
if (ex != nullptr) {
```

Reply:
I like this notation. Clearly defines a variable scope only for the if statement. This case is simple.
But another example:
```
if (auto val_a = <dynamic_cast<A *>(x)) {
    // here is val_a defined
} else if (auto val_b = <dynamic_cast<B *>(x)) {
    // here is val_b defined
} else if (auto val_c = <dynamic_cast<C *>(x)) {
    // here is val_c defined
}
```
And if `<dynamic_cast<A *>(x)` is successful no further casts are performed.

If you don't like it, there is another solution, C++17 and higher allows this syntax:
`if (auto ex = dynamic_cast<const Error *>(&e); ex != nullptr)`

Reply:
Ok, valid point, let's keep it this way then. We've also agreed to keep using the implicit bool conversion for pointers, as it is a very common occurrence in C/C++ and should be well understood.